rfcs/0000-sentry-architecture.md

```diff
@@ -0,0 +1,84 @@
+## Summary
+[summary]: #summary
+
+Introduce a new "hidden" and "sentry" mode to the daemon for building
+DoS-resilient network topologies.
+
+## Motivation
+[motivation]: #motivation
+
+Block producer availability is important both for healthy consensus (blocks
+have a limited time to be broadcast before they are no longer useful) and also
+for node operator profits. Block producers (and, to a lesser extent, SNARK
+workers) need to be powerful enough to produce their proofs in a reasonable
+amount of time, and have operational security requirements around protecting
+the private key. If a DoS takes a node offline such that they can't broadcast
+their work, or put the CPUs to use, the operator loses money and consensus is
+weakened. It is also risky to directly expose a machine with sensitive key
+material to untrusted internet traffic. Sensitive nodes can be protected from
+these risks by running them in a new "hidden mode". These hidden nodes will
+communicate via nodes running in "sentry mode".
+
+The recommended way to deploy this would be to have several sentries, possibly
+in different cloud providers/datacenters, configured per hidden node.
+
+## Detailed design
+[detailed-design]: #detailed-design
+
+New RPCs:
+
+- `Sentry_forward`: sent from a hidden node to a sentry node with some new work, to be broadcast over the gossip net.
+- `Please_sentry`: sent from a hidden node to its sentries periodically as a sort of "keepalive" to ensure the sentry node continues forwarding messages even if it crashes and forgets about us. These should be sent at least once per slot.
```

Comment:
By "it crashes" you mean the `sentry node` crashes?

Reply:
Yes, the idea being that if it has crashed then it will not remember the previous `Please_sentry` requests.